docs/data-sources/aws_transit_gateway_attachment.md

```diff
@@ -0,0 +1,51 @@
+---
+page_title: "hcp_aws_transit_gateway_attachment Data Source - terraform-provider-hcp"
+subcategory: ""
+description: |-
+  The AWS Transit gateway attachment data source provides information about an existing Transit gateway attachment.
+---
+
+# Data Source `hcp_aws_transit_gateway_attachment`
+
+The AWS Transit gateway attachment data source provides information about an existing Transit gateway attachment.
+
+## Example Usage
+
+```terraform
+data "hcp_aws_transit_gateway_attachment" "test" {
+  hvn_id                        = var.hvn_id
+  transit_gateway_attachment_id = var.transit_gateway_attachment_id
+}
+```
+
+## Schema
+
+### Required
+
+- **hvn_id** (String) The ID of the HashiCorp Virtual Network (HVN).
+- **transit_gateway_attachment_id** (String) The ID of the Transit gateway attachment.
+
+### Optional
+
+- **id** (String) The ID of this resource.
```

Comment:
So this `id` represents our unique Terraform id stored in the state (format is `/project/123/hashicorp.network.tgw-attachment/456`), but the description here is not particularly descriptive, and I think it might actually be confusing for users. Note that it exists here because it is autogenerated by `go generate`.

I think we should leave it for now, because it aligns with the generator and our other resources, but I've created a ticket to see if it makes sense to modify our docs generation to not include these: https://hashicorp.atlassian.net/browse/HCPE-883